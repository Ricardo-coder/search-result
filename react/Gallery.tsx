import React from 'react'
import GalleryLayout from './GalleryLayout'
//@ts-ignore
import GalleryLegacy from './GalleryLegacy'

const Gallery: React.FC<
  GalleryLegacyProps | GalleryLayoutPropsWithSlots
> = props => {
  if ('layouts' in props && props.layouts.length > 0) {
    const {
      layouts,
      lazyItemsRemaining,
      products,
      showingFacets,
      summary,
      ...slots
    } = props as GalleryLayoutPropsWithSlots

    return (
      <GalleryLayout
        layouts={layouts}
        lazyItemsRemaining={lazyItemsRemaining}
        products={products}
        showingFacets={showingFacets}
        summary={summary}
        slots={slots}
      />
    )
  }

  return <GalleryLegacy {...(props as GalleryLegacyProps)} />
}

export default Gallery
