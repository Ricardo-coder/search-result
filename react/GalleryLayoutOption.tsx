import React, { ComponentType } from 'react'

import { useCssHandles } from 'vtex.css-handles'
import { SearchPageContext } from 'vtex.search-page-context'
import { SWITCH_GALLERY_LAYOUT_TYPE } from './constants'

const CSS_HANDLES = ['galleryLayoutOption'] as const
const { useSearchPageState, useSearchPageStateDispatch } = SearchPageContext

interface OptionSlot {
  isActive?: boolean
}

interface GalleryLayoutOptionProps {
  Option: ComponentType<OptionSlot>
  name: string
}

const GalleryLayoutOption: React.FC<GalleryLayoutOptionProps> = ({
  Option,
  name,
}) => {
  const handles = useCssHandles(CSS_HANDLES)
  const { galleryLayout } = useSearchPageState()
  const dispatch = useSearchPageStateDispatch()

  const isActive = name === galleryLayout

  const handleOptionClick = () => {
    dispatch({
      type: SWITCH_GALLERY_LAYOUT_TYPE,
      args: { galleryLayout: name },
    })
  }

  return (
    <button
      className={`${handles.galleryLayoutOption} grow dib br-100 pa2 mr2 ml2 bw0 pointer outline-0 bg-transparent`}
      onClick={handleOptionClick}
      role="radio"
      aria-checked={isActive}
      aria-controls="gallery-layout-container"
      aria-label={`Switch to ${name} layout`}
    >
      <Option isActive={isActive} />
    </button>
  )
}

export default GalleryLayoutOption
