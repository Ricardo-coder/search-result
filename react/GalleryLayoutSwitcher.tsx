import React from 'react'
import classNames from 'classnames'
import { useCssHandles } from 'vtex.css-handles'

const CSS_HANDLES = ['galleryLayoutSwitcher'] as const

const GalleryLayoutSwitcher: React.FC = ({ children }) => {
  const handles = useCssHandles(CSS_HANDLES)

  const gallerySwitcherClasses = classNames(
    handles.galleryLayoutSwitcher,
    'flex flex-row flex-wrapitems-stretch bn ph1 na4'
  )

  return (
    <div
      className={gallerySwitcherClasses}
      role="radiogroup"
      aria-label="Gallery layout switcher"
    >
      {children}
    </div>
  )
}

export default GalleryLayoutSwitcher
