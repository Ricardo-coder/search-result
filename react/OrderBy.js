import PropTypes from 'prop-types'
import React, { useMemo } from 'react'
import { injectIntl, intlShape } from 'react-intl'

import SelectionListOrderBy from './components/SelectionListOrderBy'

export const SORT_OPTIONS = [
  {
    value: 'OrderByScoreDESC',
    label: 'store/ordination.relevance',
  },
  {
    value: 'OrderByTopSaleDESC',
    label: 'store/ordination.sales',
  },
  {
    value: 'OrderByReleaseDateDESC',
    label: 'store/ordination.release.date',
  },
  {
    value: 'OrderByBestDiscountDESC',
    label: 'store/ordination.discount',
  },
  {
    value: 'OrderByPriceDESC',
    label: 'store/ordination.price.descending',
  },
  {
    value: 'OrderByPriceASC',
    label: 'store/ordination.price.ascending',
  },
  {
    value: 'OrderByNameASC',
    label: 'store/ordination.name.ascending',
  },
  {
    value: 'OrderByNameDESC',
    label: 'store/ordination.name.descending',
  },
  {
    value: 'OrderByCollection',
    label: 'store/ordination.collection.ascending',
  },
]

const OrderBy = ({
  orderBy,
  message,
  intl,
  hiddenOptions = [],
  showOrderTitle = true,
}) => {
  const sortingOptions = useMemo(() => {
    return SORT_OPTIONS.filter(
      option => !hiddenOptions.includes(option.value)
    ).map(({ value, label }) => {
      return {
        value: value,
        label: intl.formatMessage({ id: label }),
      }
    })
  }, [intl, hiddenOptions])

  return (
    <SelectionListOrderBy
      orderBy={orderBy}
      message={message}
      options={sortingOptions}
      showOrderTitle={showOrderTitle}
    />
  )
}

OrderBy.propTypes = {
  /** Which sorting option is selected. */
  orderBy: PropTypes.string,
  /** Intl instance. */
  intl: intlShape,
  /** Options to be hidden. (e.g. `["OrderByNameASC", "OrderByNameDESC"]`) */
  hiddenOptions: PropTypes.arrayOf(PropTypes.string),
  /** Message to be displayed */
  message: PropTypes.string,
  /** Show or hide order title */
  showOrderTitle: PropTypes.boolean,
}

export default injectIntl(OrderBy)
