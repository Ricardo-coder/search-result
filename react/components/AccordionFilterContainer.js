import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { injectIntl, intlShape } from 'react-intl'
import classNames from 'classnames'
import { IconCaret } from 'vtex.store-icons'
import { Spinner } from 'vtex.styleguide'

import AccordionFilterItem from './AccordionFilterItem'
import DepartmentFilters from './DepartmentFilters'
import AccordionFilterGroup from './AccordionFilterGroup'
import AccordionFilterPriceRange from './AccordionFilterPriceRange'
import { useCssHandles } from 'vtex.css-handles'

const CSS_HANDLES = [
  'filterBreadcrumbsItem',
  'filterBreadcrumbsItemName',
  'filterBreadcrumbsContent',
  'filterBreadcrumbsText',
  'filterLoadingOverlay',
]
import styles from '../searchResult.css'

const CATEGORIES_TITLE = 'store/search.filter.title.categories'

const AccordionFilterContainer = ({
  filters,
  intl,
  onFilterCheck,
  tree,
  onCategorySelect,
  priceRange,
  loading,
  onClearFilter,
  showClearByFilterOnMobile,
}) => {
  const [openItem, setOpenItem] = useState(null)
  const handles = useCssHandles(CSS_HANDLES)

  const handleOpen = id => e => {
    e.preventDefault()

    if (openItem === id) {
      setOpenItem(null)
    } else {
      setOpenItem(id)
    }
  }

  const handleKeyDown = e => {
    if (e.key === 'Enter') {
      setOpenItem(null)
    }
  }

  const nonEmptyFilters = filters.filter(spec => spec.facets.length > 0)

  const departmentsOpen = openItem === CATEGORIES_TITLE

  const itemClassName = classNames(
    styles.accordionFilterItemOptions,
    'ph5 pt3 h-100 overflow-scroll'
  )

  const loadingOverlayClasses = classNames(
    'fixed dim top-0 w-100 vh-100 left-0 z-9999 justify-center items-center justify-center items-center',
    {
      dn: !loading,
      flex: loading,
    }
  )

  return (
    <div className={classNames(styles.accordionFilter, 'h-100')}>
      <div
        className={classNames(
          styles.filterAccordionBreadcrumbs,
          'pointer flex flex-row items-center pa5 bg-base w-100 z-max bb b--muted-4'
        )}
      >
        <div
          role="button"
          tabIndex={0}
          className={`${handles.filterBreadcrumbsContent} pv4 flex items-center`}
          onClick={() => setOpenItem(null)}
          onKeyDown={handleKeyDown}
        >
          <div
            className={classNames(
              `${handles.filterBreadcrumbsText} t-heading-4`,
              {
                'c-muted-2': openItem,
                'c-on-base': !openItem,
              }
            )}
          >
            {intl.formatMessage({
              id: 'store/search-result.filter-breadcrumbs.primary',
            })}
          </div>
        </div>
        {openItem && (
          <div
            className={`${handles.filterBreadcrumbsItem} pv4 flex items-center`}
          >
            <IconCaret orientation="right" size={13} />
            <div
              className={`${handles.filterBreadcrumbsItemName} pl3 t-heading-4 c-on-base`}
            >
              {intl.formatMessage({ id: openItem })}
            </div>
          </div>
        )}
      </div>

      {tree.length > 0 && (
        <AccordionFilterItem
          title={CATEGORIES_TITLE}
          open={departmentsOpen}
          show={!openItem || departmentsOpen}
          onOpen={handleOpen(CATEGORIES_TITLE)}
          onClearFilter={onClearFilter}
        >
          <div className={itemClassName}>
            <DepartmentFilters
              tree={tree}
              isVisible={tree.length > 0}
              onCategorySelect={onCategorySelect}
              hideBorder
            />
          </div>
        </AccordionFilterItem>
      )}

      {nonEmptyFilters.map(filter => {
        const { type, title } = filter
        const isOpen = openItem === filter.title

        switch (type) {
          case 'PriceRanges':
            return (
              <AccordionFilterPriceRange
                title={filter.title}
                facets={filter.facets}
                key={title}
                className={itemClassName}
                open={isOpen}
                show={!openItem || isOpen}
                onOpen={handleOpen(title)}
                onFilterCheck={onFilterCheck}
                priceRange={priceRange}
              />
            )
          default:
            return (
              <AccordionFilterGroup
                title={filter.title}
                facets={filter.facets}
                key={title}
                className={itemClassName}
                open={isOpen}
                show={!openItem || isOpen}
                onOpen={handleOpen(title)}
                onFilterCheck={onFilterCheck}
                onClearFilter={onClearFilter}
                showClearByFilterOnMobile={showClearByFilterOnMobile}
              />
            )
        }
      })}
      <div
        style={{ background: 'rgba(3, 4, 78, 0.4)' }}
        className={classNames(
          handles.filterLoadingOverlay,
          loadingOverlayClasses
        )}
      >
        <Spinner />
      </div>
    </div>
  )
}

AccordionFilterContainer.propTypes = {
  /** Current available filters */
  filters: PropTypes.arrayOf(PropTypes.object),
  /** Intl instance */
  intl: intlShape,
  /** Filters mapped for checkbox */
  filtersChecks: PropTypes.object,
  /** Checkbox hit callback function */
  onFilterCheck: PropTypes.func,
  /** Current price range filter query parameter */
  priceRange: PropTypes.string,
  tree: PropTypes.any,
  onCategorySelect: PropTypes.func,
  loading: PropTypes.bool,
  /** Clear filter function */
  onClearFilter: PropTypes.func,
  /** Whether a clear button that clear all options in a specific filter should appear beside the filter's name (true) or not (false). */
  showClearByFilterOnMobile: PropTypes.bool,
}

export default injectIntl(AccordionFilterContainer)
