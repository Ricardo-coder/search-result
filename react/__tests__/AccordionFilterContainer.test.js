import React from 'react'
import { render } from '@vtex/test-tools/react'
import AccordionFilterContainer from '../components/AccordionFilterContainer'

describe('<AccordionFilterContainer />', () => {
  const renderComponent = () => {
    const filters = [
      {
        title: 'price-range1',
        type: 'PriceRanges',
        facets: [
          {
            name: 'name2',
            quantity: 5,
          },
        ],
      },
    ]

    const priceRange = '1 TO 20000000'
    const tree = []

    return render(
      <AccordionFilterContainer
        filters={filters}
        tree={tree}
        priceRange={priceRange}
      />
    )
  }

  it('check if mobile price range exists', () => {
    const { getByText } = renderComponent()
    expect(getByText('price-range1')).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { asFragment } = renderComponent()

    expect(asFragment()).toMatchSnapshot()
  })
})
