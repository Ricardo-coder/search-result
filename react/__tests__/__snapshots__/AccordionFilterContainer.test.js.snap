// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionFilterContainer /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="accordionFilter h-100"
  >
    <div
      class="filterAccordionBreadcrumbs pointer flex flex-row items-center pa5 bg-base w-100 z-max bb b--muted-4"
    >
      <div
        class="filterBreadcrumbsContent pv4 flex items-center"
        role="button"
        tabindex="0"
      >
        <div
          class="filterBreadcrumbsText t-heading-4 c-on-base"
        >
          Filters
        </div>
      </div>
    </div>
    <div
      class="accordionFilterContainer pl7"
    >
      <div
        class="accordionFilterItem filterAccordionItemBox t-body pr5 pv3 pointer bb b--muted-5"
        role="button"
        tabindex="0"
      >
        <div
          class="accordionFilterContent pv4 c-on-base t-heading-5"
        >
          <span
            class="accordionFilterItemTitle"
          >
            price-range1
          </span>
          <span
            class="accordionFilterItemIcon fr"
          >
            <div>
              IconCaret
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
